modules/extract/functions/EXTRACT/fn_EXTRACT_Extract.sqf

```diff
@@ -2,36 +2,37 @@
 
 if (!isServer) exitWith {};
 
-private _class = _this select 0;
-private _spawnLoc = _this select 1;
-private _alt = _this select 2;
-private _landLoc = _this select 3;
-private _waitFor = _this select 4;
-private _extractLoc = _this select 5;
-private _vehInit = nil;
+params [
+    ["_class", "", [""]],
+    ["_spawnLoc", ""],
+    ["_alt", 15, [15]],
+    ["_landLoc", ""],
+    ["_waitFor", blufor],
+    ["_extractLoc", ""],
+    ["_vehInit", "", [""]]
+];
 
 private _spawnPos = nil;
 private _landPos = nil;
 private _extractPos = nil;
 private _isHelicopter = false;
 
-if (count _this > 6) then {
-  _vehInit = _this select 6;
-};
-
-switch (typeName _spawnLoc) do {
-  case "STRING": {_spawnPos = getMarkerPos _spawnLoc};
-  case "OBJECT": {_spawnPos = getPos _spawnLoc};
+if ("" isEqualType _spawnLoc) then {
```

Comment:
```sqf
private _spawnPos = if (_spawnLoc isEqualType "") then {
  getMarkerPos _spawnLoc
} else {
  getPos _spawnLoc
};
```

then you can remove the nil references for the vars above

Reply:
Thanks that's a good idea!